thirdparty/patchmatchnet/models/module.py

```diff
@@ -138,8 +138,8 @@ def differentiable_warping(
 
     Args:
         src_fea: [B, C, H, W] source features, for each source view in batch
-        src_proj: [B, 4, 4] source camera poses, for each source view in batch
-        ref_proj: [B, 4, 4] reference camera pose, for each ref view in batch
+        src_proj: [B, 4, 4] source projection matrices, for each source view in batch
```

Comment:
@codyly i think these must be 4x4 extrinsic matrices, not 3x4 projection matrices, if they are invertible

Reply:
here the projection matrices are not 3x4, in the original dataset script, the projection matrices are calculated by:
```
proj_mat = extrinsics.copy()
proj_mat[:3, :4] = intrinsics @ proj_mat[:3, :4]
```
So proj_mat is a 4x4 invertible matrix